readme.md

```diff
@@ -200,119 +107,244 @@ $ adb forward tcp:9222 localabstract:chrome_devtools_remote
 $ lighthouse --disable-device-emulation --disable-cpu-throttling https://mysite.com
 ```
 
-## Tests
+## Using programmatically
 
-Some basic unit tests forked are in `/test` and run via mocha. ESLint also checks for style violations.
+The example below shows how to setup and run Lighthouse programmatically as a Node module.
 
-```sh
-# lint and test all files
-npm test
+```javascript
+// Install dependencies: yarn install --dev lighthouse yargs
 
-# watch for file changes and run tests
-npm run watch
+'use strict';
 
-## run linting and unit tests separately
-npm run lint
-npm run unit
+const lighthouse = require('lighthouse');
+const ChromeLauncher = require('lighthouse/lighthouse-cli/chrome-launcher.js').ChromeLauncher;
+const Printer = require('lighthouse/lighthouse-cli/printer');
+const yargs = require('yargs');
+
+// Also could use your own custom config.
+const PERF_CONFIG = require('lighthouse/lighthouse-core/config/perf.json');
+// const DEFAULT_CONFIG = require('lighthouse/lighthouse-core/config/default.json');
+
+const _SIGINT = 'SIGINT';
+const _SIGINT_EXIT_CODE = 130;
+
+function handleError() {
+  console.error(err);
+}
+
+// Wrapper to launch Chrome and run Lighthouse.
+class LighthouseRunner {
+  constructor(url, flags, config) {
+    this.url = url;
+    this.flags = flags;
+    this.config = config;
+  }
+
+  launchChrome() {
+    this.launcher = new ChromeLauncher({
+      port: this.flags.port,
+      autoSelectChrome: !this.flags.selectChrome,
+    });
+
+    // Kill spawned Chrome process in case of ctrl-C.
+    process.on(_SIGINT, () => {
+      this.launcher.kill().then(() => process.exit(_SIGINT_EXIT_CODE), handleError);
+    });
+
+    return this.launcher.isDebuggerReady().catch(() => {
+      if (this.flags.skipAutolaunch) {
+        return;
+      }
+      return this.launcher.run();
+    });
+  }
+
+  run() {
+    return this.launchChrome()
+      .then(() => lighthouse(this.url, this.flags, this.config)) // Run Lighthouse.
+      .then(results => this.launcher.kill().then(() => results)) // Kill Chrome and return results.
+      .catch(err => {
+        // Kill Chrome if there's an error.
+        return this.launcher.kill().then(() => {
+          throw err;
+        }, handleError);
+      });
+  }
+
+  // Example of extracting the overall score from the results.
+  getOverallScore(results) {
+    const scoredAggregations = results.aggregations.filter(a => a.scored);
+    const total = scoredAggregations.reduce((sum, aggregation) => sum + aggregation.total, 0);
+    return (total / scoredAggregations.length) * 100;
+  }
+}
+
+const flags = yargs
```

Comment:
yargs especially seems like it could be cut. 9 times out of 10 it seems like people will be calling lighthouse within an existing script or with hardcoded values